### PR TITLE
[MHV-44568] Updated Navigation after sending a message

### PR DESCRIPTION
**Pull request by vsaleem**
Summary

MHV-44568: Updated Navigation after User sends a Secure Message

User will be redirected to the original folder from where the message was sent from.
Originally the user was sent to the Inbox or Sent folder regardless of where the message was sent from.

MHV-44474: User Unable to Remove Custom Folder

When a user removes an EMPTY folder they will be presented with a modal to delete the folder.
When a user attempts to remove a folder with messages, they will be present with a modal that tells them to empty the folder before they can delete it.

Related issue(s)

https://jira.devops.va.gov/browse/MHV-44568
https://jira.devops.va.gov/browse/MHV-44474

Testing done
Unit tests added
QA testing completed

Screenshots
N/A

What areas of the site does it impact?

VA.GOV - Secure Messaging

Acceptance criteria

Quality Assurance & Testing

 I fixed|updated|added unit tests and integration tests for each feature (if applicable).
 No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
 Linting warnings have been addressed
 Documentation has been updated ([link to documentation](https://github.com/department-of-veterans-affairs/vets-website/compare/44568-2nd-User-is-able-to-navigate-after-sending-a-secure-message-mhv?expand=1#) *if necessary)
 Screenshot of the developed feature is added
 [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed
Error Handling

 Browser console contains no warnings or errors.
 Events are being sent to the appropriate logging solution
 Feature/bug has a monitor built into Datadog or Grafana (if applicable)
Authentication

 Did you login to a local build and verify all authenticated routes work as expected with a test user
⚠️ Team Sites (only applies to modifications made to the VA.gov header) ⚠️

 The vets-website header does not contain any web-components
 I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
 I reached out in the #sitewide-public-websites Slack channel for questions
Requested Feedback

(OPTIONAL) What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?

### CODE REVIEW

**Review by elwoodva — APPROVED** [review]

**Review by alex-morgun-1 — CHANGES_REQUESTED** [review]
Overall looks good. I really think we need to add unit test coverage for that logic when deleting a non-empty folder

File: src/applications/mhv/secure-messaging/components/ComposeForm/ComposeForm.jsx [hunk 1]
```diff
@@ -158,11 +161,13 @@ const ComposeForm = props => {
           sendData.append('message', JSON.stringify(messageData));
           attachments.map(upload => sendData.append('uploads[]', upload));
           dispatch(sendMessage(sendData, true))
-            .then(() => history.push(Paths.INBOX))
+            .then(() =>
+              navigateToFolderByFolderId(currentFolder?.folderId || 0, history),
```
[the remark this answers is not in the document]
Comment (vsaleem) [reply src/applications/mhv/secure-messaging/components/ComposeForm/ComposeForm.jsx]: This is updated

File: src/applications/mhv/secure-messaging/components/ComposeForm/ComposeForm.jsx [hunk 2]
```diff
@@ -158,11 +161,13 @@ const ComposeForm = props => {
...
+            )
             .catch(setSendMessageFlag(false));
         } else {
           dispatch(sendMessage(JSON.stringify(messageData), false)).then(() =>
-            history.push(Paths.INBOX),
+            navigateToFolderByFolderId(currentFolder?.folderId || 0, history),
```
[the remark this answers is not in the document]
Comment (vsaleem) [reply src/applications/mhv/secure-messaging/components/ComposeForm/ComposeForm.jsx]: This is updated. Good suggestion.

File: src/applications/mhv/secure-messaging/components/ManageFolderButtons.jsx [hunk 3]
```diff
@@ -65,7 +65,7 @@ const ManageFolderButtons = () => {
 
   const openDelModal = () => {
     dispatch(closeAlert());
-    if (messages?.length > 0) {
+    if (threads.threadList.length > 0) {
```
[the remark this answers is not in the document]
Comment (vsaleem) [reply src/applications/mhv/secure-messaging/components/ManageFolderButtons.jsx:68]: I'm going to make a separate PR for this specific test.  I'm having trouble mocking it so I need to make it a side-task.

File: src/applications/mhv/secure-messaging/components/ManageFolderButtons.jsx [hunk 4]
```diff
@@ -177,10 +177,14 @@ const ManageFolderButtons = () => {
...
-            text="Remove"
+            text="Yes, remove this folder"
             onClick={confirmDelFolder}
           />
-          <va-button secondary="true" text="Cancel" onClick={closeDelModal} />
+          <va-button
+            secondary="true"
+            text="No, keep this folder"
+            onClick={closeDelModal}
+          />
```
[the remark this answers is not in the document]
Comment (vsaleem) [reply src/applications/mhv/secure-messaging/components/ManageFolderButtons.jsx]: This is also updated.

**Review by alex-morgun-1 — APPROVED** [review]

**Pull request merged**